tests/unit/api/objectDelete.js

```diff
@@ -371,3 +374,59 @@ describe('objectDelete API', () => {
         });
     });
 });
+
+
+describe('objectDelete API with versioning', () => {
+    let testPutObjectRequest;
+
+    beforeEach(() => {
+        cleanup();
+        testPutObjectRequest = new DummyRequest({
+            bucketName,
+            namespace,
+            objectKey,
+            headers: {},
+            url: `/${bucketName}/${objectKey}`,
+        }, postBody);
+
+        sinon.stub(services, 'deleteObject').callsFake(originalDeleteObject);
+        sinon.spy(metadataswitch, 'putObjectMD');
+        sinon.spy(metadataswitch, 'deleteObjectMD');
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    const testBucketPutRequest = new DummyRequest({
+        bucketName,
+        namespace,
+        headers: {},
+        url: `/${bucketName}`,
+    });
+    const testDeleteRequest = new DummyRequest({
+        bucketName,
+        namespace,
+        objectKey,
+        headers: {},
+        url: `/${bucketName}/${objectKey}`,
+    });
+
+    it('should store modified originOp when moving object null version', done => {
```

Comment:
maybe better describe the scenario, to explain this happens when putting a delete marker on top of a "master-only null version"


```suggestion
    it('should store modified originOp when storing null object due to object marker', done => {
```

...or we could make this more general, and consider this test actually tests this overall process of "upgrading" the null-version object (in case an object marker is put on top of it) : i.e. we do not just want to verify that the object has the proper originOp, but also that it has all proper metadata and "null" version id (which is mostly what you do here, actually)

Reply:
OK, clarified the test name accordingly.